scripts/vlsi_mem_gen

```diff
@@ -167,10 +167,15 @@ def gen_mem(name, width, depth, mask_gran, mask_seg, ports):
     prefix = 'RW%d_' % idx
     sequential.append('always @(posedge %sclk)' % prefix)
     sequential.append("  if (%sen && %swmode) begin" % (prefix, prefix))
-    for i in range(mask_seg):
-      mask = ('if (%swmask[%d]) ' % (prefix, i)) if pid in maskedports else ''
-      ram_range = '%d:%d' % ((i+1)*mask_gran-1, i*mask_gran)
-      sequential.append("    %sram[%saddr][%s] <= %swdata[%s];" % (mask, prefix, ram_range, prefix, ram_range))
+    if mask_seg > 0:
```

Comment:
Is it actually possible for mask_seg to be less than 1?  If it's not supposed to be, I think this should be an assertion rather than an `if`.

Reply:
Well, because the previous one did `for i in range (mask_seg)` I essentially wanted to only do this if maskseg is nonzero. I could just do `if maskseg`

Reply:
I see, no it's fine